src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/WebRequestSession.cs

```diff
@@ -105,8 +158,134 @@ public WebRequestSession()
         internal void AddCertificate(X509Certificate certificate)
         {
             Certificates ??= new X509CertificateCollection();
+            if (!Certificates.Contains(certificate))
+            {
+                ResetClient();
+                Certificates.Add(certificate);
+            }
+        }
+
+        internal virtual HttpClient GetHttpClient(bool handleRedirect, out bool clientWasReset)
+        {
+            // This indicates GetResponse will handle redirects.
+            SetStructVar(ref _allowAutoRedirect, !(handleRedirect || MaximumRedirection == 0));
```

Comment:
Please make the comment better.

Reply:
Comment updated - this comment makes less sense now the code has been moved from the webrequest cmdlet base class. handleRedirect is also reverse of what it actually does, so I've renamed that here to doNotRedirect and added documentation on the class.